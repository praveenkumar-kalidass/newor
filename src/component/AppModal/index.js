import AppModal from './AppModal';
import AppModalBody from './AppModalBody';
import AppModalFooter from './AppModalFooter';
import AppModalHeader from './AppModalHeader';

AppModal.Body = AppModalBody;
AppModal.Footer = AppModalFooter;
AppModal.Header = AppModalHeader;

export default AppModal;
