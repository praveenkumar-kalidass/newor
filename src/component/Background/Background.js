import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Polygon, Svg } from 'react-native-svg';

import COLOR from '../../constant/color';
import BackgroundContainer from './Background.style';

const Background = ({ height, width }) => {
  const primaryBackground = useMemo(
    () => `0,0 0,${height} ${width},${height} 30,${height - 30} 0,0`,
    [height, width],
  );
  const secondaryBackground = useMemo(
    () => `0,0 ${width},0 ${width},${height} ${width - 30},30 0,0`,
    [height, width],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Polygon fill={COLOR.LIGHT_PRIMARY_100} points={primaryBackground} />
        <Polygon fill={COLOR.LIGHT_SECONDARY_100} points={secondaryBackground} />
      </Svg>
    </BackgroundContainer>
  );
};

Background.propTypes = {
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
};

export default Background;
