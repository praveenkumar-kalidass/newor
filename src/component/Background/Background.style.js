import styled from 'styled-components/native';

export default styled.View`
  background-color: ${({ theme }) => theme.color.BACKGROUND_100};
  position: absolute;
`;
