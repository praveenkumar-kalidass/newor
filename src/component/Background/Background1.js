import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { Polygon, Svg } from 'react-native-svg';

import useTheme from 'theme/useTheme';
import BackgroundContainer from './Background.style';

const Background1 = ({ height, width }) => {
  const theme = useTheme();
  const primaryBackground = useMemo(
    () => `0,0 0,${height} ${width},${height} 30,${height - 30} 0,0`,
    [],
  );
  const secondaryBackground = useMemo(
    () => `0,0 ${width},0 ${width},${height} ${width - 30},30 0,0`,
    [],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Polygon fill={theme.color.PRIMARY_100} points={primaryBackground} />
        <Polygon fill={theme.color.SECONDARY_100} points={secondaryBackground} />
      </Svg>
    </BackgroundContainer>
  );
};

Background1.propTypes = {
  height: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
};

export default Background1;
