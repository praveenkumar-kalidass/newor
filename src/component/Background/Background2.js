import React, { useMemo } from 'react';
import { useWindowDimensions } from 'react-native';
import Svg, { Path } from 'react-native-svg';

import COLOR from '../../constant/color';
import BackgroundContainer from './Background.style';

const Background2 = () => {
  const { height, width } = useWindowDimensions();

  const primaryDataPoint = useMemo(
    () => `M0 0 L0 ${height / 2} L${width / 2} 0 L0 0 Z`,
    [],
  );

  const secondaryDataPoint = useMemo(
    () => `M${width} ${height} L${width / 2} ${height} L${width} ${height / 2} L${width} ${height} Z`,
    [],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Path d={primaryDataPoint} fill={COLOR.LIGHT_PRIMARY_100} />
        <Path d={secondaryDataPoint} fill={COLOR.LIGHT_SECONDARY_100} />
      </Svg>
    </BackgroundContainer>
  );
};

export default Background2;
