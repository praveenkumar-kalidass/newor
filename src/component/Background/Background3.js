import React, { useMemo } from 'react';
import { useWindowDimensions } from 'react-native';
import { Path, Svg } from 'react-native-svg';

import useTheme from 'theme/useTheme';
import BackgroundContainer from './Background.style';

const Background3 = () => {
  const { height, width } = useWindowDimensions();
  const theme = useTheme();

  const dataPoint = useMemo(
    () => `M 0,0
      L 0,${height / 5}
      C 50 ${(height / 5) + 50}, ${width - 50} ${(height / 5) + 50}, ${width} ${height / 5}
      L ${width},0
      Z`,
    [],
  );

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Path d={dataPoint} fill={theme.color.SECONDARY_100} />
      </Svg>
    </BackgroundContainer>
  );
};

export default Background3;
