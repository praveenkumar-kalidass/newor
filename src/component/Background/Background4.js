import React from 'react';
import { useWindowDimensions } from 'react-native';
import { Circle, Svg } from 'react-native-svg';

import useTheme from 'theme/useTheme';
import BackgroundContainer from './Background.style';

const Background4 = () => {
  const { height, width } = useWindowDimensions();
  const theme = useTheme();

  return (
    <BackgroundContainer>
      <Svg height={height} width={width}>
        <Circle
          cx={0}
          cy={height}
          r={250}
          stroke={theme.color.SECONDARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
        <Circle
          cx={0}
          cy={height}
          r={150}
          stroke={theme.color.SECONDARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
        <Circle
          cx={width}
          cy={0}
          r={250}
          stroke={theme.color.PRIMARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
        <Circle
          cx={width}
          cy={0}
          r={150}
          stroke={theme.color.PRIMARY_100}
          strokeWidth={25}
          fill={theme.color.BACKGROUND_100}
        />
      </Svg>
    </BackgroundContainer>
  );
};

export default Background4;
