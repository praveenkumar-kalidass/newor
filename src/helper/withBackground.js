/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';

const withBackground = ({
  Component,
  Background,
}) => (props) => (
  <>
    <Background />
    <Component {...props} />
  </>
);

export default withBackground;
