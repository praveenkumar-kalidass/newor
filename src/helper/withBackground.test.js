import React from 'react';
import { View } from 'react-native';

import { Background1 } from 'component/Background';
import withBackground from './withBackground';

describe('withBackground', () => {
  const mockComponent = () => <View />;

  it('should wrap with specific pattern, when pattern is specified', () => {
    const container = withBackground({
      Component: mockComponent,
      Background: Background1,
    });

    expect(container).toMatchSnapshot();
  });
});
