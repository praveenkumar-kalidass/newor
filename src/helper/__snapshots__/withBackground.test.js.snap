// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withBackground should wrap with specific pattern, when pattern is specified 1`] = `[Function]`;
